library/std/src/thread/mod.rs

```diff
@@ -699,16 +699,18 @@ where
 
 thread_local! {
     static CURRENT: OnceCell<Thread> = const { OnceCell::new() };
+    static CURRENT_ID: Cell<Option<ThreadId>> = const { Cell::new(None) };
```

Comment:
So now we redundantly store the thread ID twice, once in the `Thread` and once here?

At least we should explicitly document this invariant. Ideally we can avoid the redundancy...

Reply:
I'm not sure the redundancy can be avoided, seeing as we want to access the thread local without accessing the `Thread` and vice versa. And since the `Thread` instance can outlive the thread itself, we also can't store a reference to the thread local in the `Thread` instance. Maybe we could store something like a `UnsendUnsync<&'static Thread>` in a thread local pointing to the current `Thread` instance? That way we could at least avoid the `Arc` clone + drop that occurs with `thread::current().id()`...